### PR TITLE
fix(storage): checksum BlockType in block header

### LINKED ISSUE #247: storage: checksum `BlockType` in block header
Currently, we store the `BlockType` in header of each block. Therefore, this part is not checksumed. We should get this part checksumed.

### PR DESCRIPTION
**Pull request by unconsolable**
What's changed:
- Split the encode of `BlockHeader` to checksum and non-checksum
- Take non-checksum part of header (currently BlockType) to checksum calculation
- Move `BlockHeader` to the footer of block
- Change checksum validation logic

close #247 

Signed-off-by: unconsolable <chenzhipeng2012@gmail.com>

### CODE REVIEW

File: src/storage/secondary/column.rs [hunk 1]
```diff
@@ -195,18 +196,17 @@ impl Column {
                 "block is smaller than header size",
             ));
         }
-        let mut header = &block[..BLOCK_HEADER_SIZE];
-        let block_data = &block[BLOCK_HEADER_SIZE..];
+        let mut header = &block[block.len() - BLOCK_HEADER_SIZE..];
```
Comment (skyzh) [inline src/storage/secondary/column.rs:199]: should BLOCK_HEADER_SIZE be changed to BLOCK_FOOTER_SIZE?

**Review by skyzh — APPROVED** [review]
Rest LGTM. If you don't want to resolve the comments for now, ping me for merging and I'll do the rest.

File: src/storage/secondary/column.rs (hunk 1 above)
Comment (unconsolable) [reply src/storage/secondary/column.rs:199]: Shall the struct name `BlockHeader` be renamed as `BlockFooter` as well?

Comment (skyzh) [reply src/storage/secondary/column.rs:199]: Yes, +1 for this!

Comment (skyzh) [reply src/storage/secondary/column.rs:199]: Or BlockMeta, whatever you want.

**Pull request merged**